lib/storage/keyval/authorities.go

```diff
@@ -162,3 +161,29 @@ func (b *backend) DeactivateCertAuthority(id teleservices.CertAuthID) error {
 
 	return nil
 }
+
+// CompareAndSwapCertAuthority updates the cert authority value if the existing
+// value matches existing parameter
+func (b *backend) CompareAndSwapCertAuthority(new, existing teleservices.CertAuthority) error {
+	if err := new.Check(); err != nil {
+		return trace.Wrap(err)
+	}
+	newData, err := teleservices.GetCertAuthorityMarshaler().MarshalCertAuthority(new)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	existingData, err := teleservices.GetCertAuthorityMarshaler().MarshalCertAuthority(existing)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	ttl := b.ttl(new.Expiry())
+	var outData []byte
```

Comment:
Is `outData` used anywhere?

Reply:
No, but I don't think I can pass `nil` instead of it and I wanted to keep the interface of `compareAndSwapBytes` similar to existing `compareAndSwap`.